smartsim/launchable/job.py

```diff
@@ -50,26 +56,44 @@ def __init__(
         self,
         entity: SmartSimEntity,
         launch_settings: LaunchSettings,
+        name: str | None = None,
     ):
         super().__init__()
         self._entity = deepcopy(entity)
         self._launch_settings = deepcopy(launch_settings)
-        # TODO: self.warehouse_runner = JobWarehouseRunner
+        self._name = name if name else entity.name
+        check_name(self._name)
+
+    @property
+    def name(self) -> str:
+        """Retrieves the name of the Job."""
+        return self._name
+
+    @name.setter
+    def name(self, name: str) -> None:
+        """Sets the name of the Job."""
+        check_name(name)
+        logger.info(f'Overwriting the Job name from "{self._name}" to "{name}"')
```

Comment:
Nit: this seems more like a `debug` message IMO
```suggestion
        logger.debug(f'Overwriting the Job name from "{self._name}" to "{name}"')
```